src/test/kotlin/no/nav/tilleggsstonader/sak/vilkår/stønadsvilkår/regler/evalutation/OppdaterVilkårTest.kt

```diff
@@ -33,48 +41,202 @@ internal class OppdaterVilkårTest {
+                assertThatThrownBy {
+                    validerVilkårOgBeregnResultat(vilkår, opprettVilkårDto.copy(utgift = null))
+                }.hasMessageContaining("Mangler utgift på vilkår")
+            }
+
+            @Test
+            fun `skal ikke kaste feil hvis ikke oppfylt pass av barn ikke inneholder beløp`() {
+                val dto = opprettVilkårDto.copy(utgift = null, delvilkårsett = ikkeOppfylteDelvilkårPassBarnDto())
+                validerVilkårOgBeregnResultat(vilkår, dto)
+            }
         }
 
-        @Disabled // TODO fiks når annen type enn eksempel er tilgjengelig
-        @Test
-        fun `skal validere at annet vilkår enn pass av barn ikke inneholder beløp`() {
-            assertThatThrownBy {
-                validerVilkårOgBeregnResultat(vilkår, opprettVilkårDto.copy(vilkårType = VilkårType.EKSEMPEL))
-            }.hasMessageContaining("Mangler beløp på vilkår")
+        @Nested
+        inner class Boutgifter {
+            @Nested
+            inner class MidlertidigOvernatting {
+                val opprettVilkårDto =
+                    OpprettVilkårDto(
+                        vilkårType = VilkårType.MIDLERTIDIG_OVERNATTING,
+                        behandlingId = behandlingId,
+                        delvilkårsett = oppfylteDelvilkårMidlertidigOvernattingDto(),
+                        fom = LocalDate.now(),
+                        tom = LocalDate.now().plusDays(1),
+                        utgift = 1,
+                        erNullvedtak = false,
+                    )
+                val vilkår =
+                    vilkår(
+                        behandlingId = behandlingId,
+                        type = VilkårType.MIDLERTIDIG_OVERNATTING,
+                        delvilkår = oppfylteDelvilkårMidlertidigOvernatting(),
+                    )
+
+                @Test
+                fun `skal validere at man har med fom og tom for vilkår for midlertidig overnatting`() {
+                    assertThatThrownBy {
+                        validerVilkårOgBeregnResultat(vilkår, opprettVilkårDto.copy(fom = null))
+                    }.hasMessageContaining("Mangler fra og med/til og med på vilkår")
+
+                    assertThatThrownBy {
+                        validerVilkårOgBeregnResultat(vilkår, opprettVilkårDto.copy(tom = null))
+                    }.hasMessageContaining("Mangler fra og med/til og med på vilkår")
+                }
+
+                @Test
+                fun `skal kaste feil hvis innvilget midlertidig overnatting ikke inneholder beløp`() {
+                    assertThatThrownBy {
+                        validerVilkårOgBeregnResultat(vilkår, opprettVilkårDto.copy(utgift = null))
+                    }.hasMessageContaining("Mangler utgift på vilkår")
+                }
+
+                @Test
+                fun `skal ikke kaste feil hvis ikke oppfylt midlertidig overnatting ikke inneholder beløp`() {
```

Comment:
Tre negasjoner i samme setning 🙀
```suggestion
                fun `utgift kan mangle når midlertidig overnatting ikke er oppfylt `() {
```

Reply:
En salgs rekord? Takk - fikser 🙏